### PR TITLE
Vuejs neomad

### PR DESCRIPTION
**Pull request by vinyll**
This PR is moving from defining styles in css to defining styles into components.

The components may then be called directly (ex: `<x-button>`) or transform a regular node (ex: `<a is=x-button>`).

It's currently implementing 4 elements:
- `x-button`
- `x-avatar`
- `x-form` (with a `class=login` behavior)
- `x-article` (with a `class=preview` behavior)

---

1. on utilise un composant, par exemple le menu, comme ça : https://github.com/neomad-team/styleguide/pull/10/commits/1d5d1ca6e6143dc9c02d6687b85fc466322dc13b#diff-e68a9f592ca9705e8a3760d94f00ad1bR65
2. le tag `<x-menu>` est rendu disponible parce qu'on le déclare dans https://github.com/neomad-team/styleguide/pull/10/commits/1d5d1ca6e6143dc9c02d6687b85fc466322dc13b#diff-1c90ff38a08209f9ebd4d05d1e43358eR12
3. le composant en question (css et js) est dispo dans le fichier _x-menu.vue_: https://github.com/neomad-team/styleguide/pull/10/commits/1d5d1ca6e6143dc9c02d6687b85fc466322dc13b#diff-8afd2c20ecf868ef9d993c30a1732904R1 (edited)

### CODE REVIEW

File: demo/atoms.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  <x-avatar href=/test/me/>
+    <img alt="John Doe" src=https://neomad.org/static/uploads/avatars/5914dbdca193770016e289a7>
+  </x-avatar>
+</div>
+
+<a href=components.html>View components</a>
+
+
+<script src=/dist/build.js></script>
```
Comment (fuentesloic) [inline demo/atoms.html:27]: `src` should be `../dist/build.js`

File: demo/components.html [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      <a href="/@vinyll/">Profile</a>
+      <a href="/privacy/">Privacy settings</a>
+      <a href="/logout/">Logout</a>
+    </nav>
+  </x-menu>
+
+
+</div>
+
+<script src=/dist/build.js></script>
```
Comment (fuentesloic) [inline demo/components.html:79]: `src`should be `../dist/build.js`

**Review by fuentesloic — CHANGES_REQUESTED** [review]

File: demo/components.html [hunk 3]
```diff
@@ -36,10 +36,10 @@
 
   <h2>Article preview</h2>
 
-  <article class=preview>
+  <x-article class=preview>
     <header>
-      <a class=avatar href=/@john-doe/>
-        <img src=https://neomad.org/static/uploads/avatars/5914dbdca193770016e289a7>
+      <a href=/@john-doe/>
+        <x-avatar src=https://neomad.org/static/uploads/avatars/5914dbdca193770016e289a7></x-avatar>
```
Comment (fuentesloic) [inline demo/components.html]: should be `<img is=x-avatar ...`

File: demo/components.html [hunk 4]
```diff
@@ -57,7 +57,23 @@
         Sujet de conférence. Ces 2 derniers jours – comme chaque année à cette période – a eu lieu la rencontre MiXiT. Au-delà de la conférence, ce qui fait la beauté de cet évènement, ce sont les...
       </a>
     </section>
-  </article>
+  </x-article>
+
+
+  <h2>Menu</h2>
+
+  <x-menu>
```
Comment (fuentesloic) [inline demo/components.html:65]: same here should be `<nav is=x-menu>`

File: demo/components.html [hunk 5]
```diff
@@ -57,7 +57,23 @@
...
+    <a href="/articles/" title="Read Neomad's adventures">Read</a>
+    <a href="/around/">Around me</a>
+    <x-avatar src=/static/uploads/avatars/5914dbdca193770016e289a7 alt=""></x-avatar>
```
Comment (fuentesloic) [inline demo/components.html:68]: again with `<img is=x-avatar ...`

**Comment by vinyll** [comment]
I'm not sure about which approach to adopt for the demo folder.
Here I went for using the explicit tag name for easier comprehension, keeping in mind that for SEO purpose the components would probably be used with a html5 template and the `is=…` attribute.

Opinions are welcome regarding this.

**Comment by vinyll** [comment]
Because the `/demo` folder and the `/dist` folder are totally decoupled (at the opposite of a `/static` folder that would hold all the js/css/images/fonts/…) I'd rather stick with an absolute path in this case. We may discuss about that.

**Pull request closed without merging**